src/lib.rs

```diff
@@ -442,3 +442,20 @@ storage_types! {
         }
     }
 }
+
+/// Shared tools for parsing quantities from strings.
+pub mod str {
+    /// Represents an error encountered while parsing a string into a `Quantity`.
+    #[derive(Clone, Copy, Debug)]
+    pub enum ParseQuantityError {
+        /// No separators (spaces) were encountered.
+        NoSeparator,
+        /// An error occured while parsing the value (first) portion of the string.
+        ///
+        /// Due to exhaustiveness and type system limitations, this variant does not encode
+        /// the underlying parse error.
+        ValueParseError,
+        /// The abbreviation used wasn't found for this quantity.
+        UnknownAbbreviation,
```

Comment:
Consider changing to `UnknownUnit` and note in the documentation that currently only the abbreviation is checked but future updates will include the description.